### ISSUE TITLE
How to test a vue component write by tsx

### ISSUE THREAD

**Issue by Hyphon** [vitest-dev/vitest#369]
### Describe the bug

throw errors when try to test a tsx component.

deps:
```
"vue": "^3.2.25"
"@vitejs/plugin-vue": "^2.0.1",
"@vitejs/plugin-vue-jsx": "^1.3.3",
"@vue/test-utils": "^2.0.0-rc.18",
"happy-dom": "*",
 "vitest": "0.0.122"
```

error:
```
TypeError: Cannot read properties of undefined (reading 'modules')
> callWithErrorHandling ../../node_modules/.pnpm/@vue+runtime-core@3.2.26/node_modules/@vue/runtime-core/dist/runtime-core.esm-bundler.js:6737:22
```

![image](https://user-images.githubusercontent.com/11419600/147727018-c360b2c1-3fc2-48c1-bde6-417d4d9e7ed9.png)

### Reproduction

Component:
```
// Case.tsx
import { defineComponent, ref, watchEffect } from "vue";

export default defineComponent({
  name: "TestComponent",
  props: {
    value: String,
  },
  emits: ["update:value"],
  setup(props, { emit }) {
    const local = ref("");

    watchEffect(() => {
      emit("update:value", local);
    });
    watchEffect(() => {
      local.value = props.value;
    });

    return {
      local,
    };
  },
  render() {
    return (
      <a-select v-model={[this.local, "value"]}>
        <a-select-option value="aaa">aaa</a-select-option>
      </a-select>
    );
  },
});
```

case.test.ts
```typescript
import Case from "../components/Case";
import { shallowMount } from "@vue/test-utils";

test("mount component", () => {
  const wrapper = shallowMount(Case, {
    props: {
      value: "test",
    },
    global: {
      stubs: ["a-select", "a-select-option"],
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

```

### System Info

```shell
System:
    OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (3) x64 Intel(R) Core(TM) i5-9400 CPU @ 2.90GHz
    Memory: 1.14 GB / 7.77 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 16.13.1 - /home/packages/node/bin/node
    Yarn: 1.22.17 - /home/packages/node/bin/yarn
    npm: 8.1.2 - /home/packages/node/bin/npm
  npmPackages:
    vite: ^2.7.9 => 2.7.9 
    vitest: workspace:* => 0.0.122
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Can you make a reproduction on https://stackblitz.com/ ?

**Comment by Hyphon** [comment]
> Can you make a reproduction on https://stackblitz.com/ ?

https://stackblitz.com/edit/vue-vitest-tsx

**Comment by sheremet-va** [comment]
Fixed in [v0.0.139](https://github.com/vitest-dev/vitest/releases/tag/v0.0.139)

Please add this to your test config:

```ts
import { defineConfig } from 'vite'

 export default defineConfig({
   test: {
     transformMode: {
       web: [/\.[jt]sx$/],
     },
   },
 })

```
